torch_geometric/data/hetero_data.py

```diff
@@ -437,6 +437,79 @@ def rename(self, name: NodeType, new_name: NodeType) -> 'HeteroData':
 
         return self
 
+    def subgraph(self, subset: Dict[NodeType, Tensor]) -> 'HeteroData':
+        r"""Returns the induced subgraph containing the node types and
+        corresponding nodes in :obj:`subset`.
+
+        .. code-block:: python
+
+            data = HeteroData()
+            data['paper'].x = (...)
+            data['author'].x = (...)
+            data['conference'].x = (...)
+            data['paper', 'cites', 'paper'].edge_index = (...)
+            data['author', 'paper'].edge_index = (...)
+            data['paper', 'conference'].edge_index = (...)
+            print(data)
+            >>> HeteroData(
+                paper={ x=[10, 16] },
+                author={ x=[5, 32] },
+                conference={ x=[5, 8] },
+                (paper, cites, paper)={ edge_index=[2, 50] },
+                (author, to, paper)={ edge_index=[2, 30] },
+                (paper, to, conference)={ edge_index=[2, 25] }
+                )
+
+            subset ={'paper': torch.tensor([3,4,5,6]),
+                     'author': torch.tensor([0,2])}
+
+            print(data.subgraph(subset))
+            >>> HeteroData(
+                paper={ x=[4, 16] },
+                author={ x=[2, 32] },
+                (paper, cites, paper)={ edge_index=[2, 24] },
+                (author, to, paper)={ edge_index=[2, 5] }
+                )
+
+        Args:
+            subset (Dict[str, Tensor or BoolTensor]): A dictonary holding the
+                nodes to keep for each node type.
+        """
+
+        data = HeteroData(self._global_store)
+
+        for node_type in self.node_types:
+            if node_type not in subset:
+                continue
+            cur_num_nodes = self[node_type].num_nodes
+            for key, value in self[node_type].items():
+                if isinstance(value,
```

Comment:
maybe you could use the `is_node_attr` and `is_edge_attr` to simplify this loop quite a bit?